pkgs/applications/misc/nnn/default.nix

```diff
@@ -1,48 +1,97 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, makeWrapper, ncurses, readline
-, archivemount, atool, fzf, libarchive, rclone, sshfs, unzip, vlock
-, conf ? null, withIcons ? false, withNerdIcons ? false }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, pkg-config
+, symlinkJoin
+, file
+, ncurses
+, readline
+, which
+# optional dependencies
+#   these can alternatively be picked up from PATH at runtime
+, withExtraTools ? false
+, archivemount
+, atool
+, fzf
+, libarchive
+, rclone
+, sshfs
+, unzip
+, vlock
+# options
+, conf ? null
+, withIcons ? false
+, withNerdIcons ? false
+}:
 
 # Mutually exclusive options
 assert withIcons -> withNerdIcons == false;
 assert withNerdIcons -> withIcons == false;
 
-stdenv.mkDerivation rec {
-  pname = "nnn";
-  version = "4.0";
+let
+  binPath = lib.makeBinPath [ file which ];
 
-  src = fetchFromGitHub {
-    owner = "jarun";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "0cbxgss9j0bvsp3czjx1kpm9id7c5xxmjfnvjyk3pfd69ygif2kl";
-  };
+  extraBinPath = lib.makeBinPath ([
+    archivemount
+    atool
+    fzf
+    libarchive
+    rclone
+    sshfs
+    unzip
+  ] ++ lib.optionals stdenv.isLinux [
+    vlock
+  ]);
+
+  nnn = stdenv.mkDerivation rec {
+    pname = "nnn";
+    version = "4.0";
+
+    src = fetchFromGitHub {
+      owner = "jarun";
+      repo = pname;
+      rev = "v${version}";
+      sha256 = "0cbxgss9j0bvsp3czjx1kpm9id7c5xxmjfnvjyk3pfd69ygif2kl";
+    };
+
+    configFile = lib.optionalString (conf != null) (builtins.toFile "nnn.h" conf);
+    preBuild = lib.optionalString (conf != null) "cp ${configFile} src/nnn.h";
+
+    nativeBuildInputs = [ pkg-config makeWrapper ];
+    buildInputs = [ readline ncurses ];
+
+    makeFlags = [ "PREFIX=${placeholder "out"}" ]
+      ++ lib.optional withIcons [ "O_ICONS=1" ]
+      ++ lib.optional withNerdIcons [ "O_NERD=1" ];
+
+    # shell completions
+    postInstall = ''
+      install -Dm555 misc/auto-completion/bash/nnn-completion.bash $out/share/bash-completion/completions/nnn.bash
+      install -Dm555 misc/auto-completion/zsh/_nnn -t $out/share/zsh/site-functions
+      install -Dm555 misc/auto-completion/fish/nnn.fish -t $out/share/fish/vendor_completions.d
```

Comment:
This could be changed to installShellCompletion.

Reply:
Fixed.